File: leetcode/100-999/118. Pascal's Triangle/solution.cpp
class Solution 
{
    public:
        vector<vector<int>> generate(int numRows) 
        {
            vector<vector<int>> pascalTriangle = {{1}};

            for (int i = 0; i < numRows - 1; ++i)
            {
                vector<int> nextRow = {1};
                
                for (int j = 0; j < pascalTriangle[i].size(); ++j)
                {
                    if (j == pascalTriangle[i].size() - 1)
                        nextRow.push_back(1);
                    else
                        nextRow.push_back(pascalTriangle[i][j] + pascalTriangle[i][j + 1]);
                }

                pascalTriangle.push_back(nextRow);
            }  

            return pascalTriangle;  
        }
};

File: leetcode/100-999/118. Pascal's Triangle/solution.java
class Solution
{
    public List<List<Integer>> generate(int numRows)
    {
        var pascalTriangle = new ArrayList<List<Integer>>();
        var firstRow = new ArrayList<Integer>();

        firstRow.add(1);
        pascalTriangle.add(firstRow);

        for (int i = 0; i < numRows - 1; ++i)
        {
            var nextRow = new ArrayList<Integer>();
            nextRow.add(1);

            for (int j = 0; j < pascalTriangle.get(i).size(); ++j)
            {
                if (j == pascalTriangle.get(i).size() - 1)
                    nextRow.add(1);
                else
                    nextRow.add(pascalTriangle.get(i).get(j) + pascalTriangle.get(i).get(j + 1));
            }

            pascalTriangle.add(nextRow);
        }

        return pascalTriangle;
    }
}